activesupport/lib/active_support/core_ext/date/conversions.rb

```diff
@@ -80,6 +80,7 @@ def readable_inspect
   #
   #   date.to_time(:utc)             # => 2007-11-10 00:00:00 UTC
   def to_time(form = :local)
+    raise ArgumentError, "Expected :local or :uts, got #{form}." unless [:local, :utc].include?(form)
```

Comment:
I took benchmark.
Add these methods to `activesupport/lib/active_support/core_ext/date/conversions.rb`

```ruby
  def to_time_with_case(form = :local)
    case
    when ![:local, :utc].include?(form)
      raise ArgumentError, "Expected :local or :utc, got #{form.inspect}."
    else
      ::Time.send(form, year, month, day)
    end
  end

  def to_time_with_rescue(form = :local)
    ::Time.send(form, year, month, day)
  rescue NoMethodError
    raise ArgumentError, "Expected :local or :utc, got #{form.inspect}."
  end
```

and create `activesupport/lib/active_support/core_ext/date/conversions.rb`

```ruby
require 'abstract_unit'
require 'active_support/time'
require 'core_ext/date_and_time_behavior'
require 'time_zone_test_helpers'
require 'benchmark/ips'

Benchmark.ips do |x|
  x.report('to_time')             { Date.new(2005, 2, 21).to_time(:utc) }
  x.report('to_time_with_case')   { Date.new(2005, 2, 21).to_time_with_case(:utc) }
  x.report('to_time_with_rescue') { Date.new(2005, 2, 21).to_time_with_rescue(:utc) }
end
```

```
$ bundle exec ruby -Itest test/core_ext/date_ext_bench_test.rb
Calculating -------------------------------------
             to_time    30.125k i/100ms
   to_time_with_case    30.181k i/100ms
 to_time_with_rescue    31.234k i/100ms
-------------------------------------------------
             to_time    389.934k (± 4.4%) i/s -      1.958M
   to_time_with_case    367.383k (± 6.7%) i/s -      1.841M
 to_time_with_rescue    391.123k (± 7.9%) i/s -      1.968M
Run options: --seed 15798

# Running:



Finished in 0.000813s, 0.0000 runs/s, 0.0000 assertions/s.

0 runs, 0 assertions, 0 failures, 0 errors, 0 skips
```

Although `to_time_with_rescue` is a bit faster than `to_time`, but I like `to_time` because it is natural checking arguments at first of method.